### PR TITLE
[WIP] owncloud provider

### PR DESCRIPTION
**Pull request by cosenal**
:construction:  DON'T MERGE, still under construction :construction: 

- [x] Basic authentication
- [x] File metadata 
- [x] Folder metadata 
- [x] Download
- [x] Upload
- [x] Delete
- [ ] Unit tests

@chrisseto @sloria  This is a very early draft. Please provide some feedback before I implement the rest.

I am cc'ing @PVince81 from ownCloud, the author of https://github.com/owncloud/pyocclient.
(Vincent: In some sense, this is a port of your library which uses [aiohttp](https://aiohttp.readthedocs.org/) instead of Requests)

Here is a test file: https://gist.github.com/cosenal/817ac522e2fd6ebe22a8
In order to test it, you need a local instance of owncloud, with user `test:test`

### CODE REVIEW

File: waterbutler/providers/owncloud/provider.py [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+
+    def __init__(self, auth, credentials, settings):
+        super().__init__(auth, credentials, settings)
+        self.auth = b64encode((credentials['username'] + ':' 
+            + credentials['password']).encode('ascii')).decode('ascii')
+        self.webdav_url = settings['base_url'] + self._DAV_URL
+ 
+    @property
+    def default_headers(self):
+        return { 'Authorization' : 'Basic %s' %  self.auth }
```
Comment (chrisseto) [inline waterbutler/providers/owncloud/provider.py]: Use new style string formatting here `'{}'.format(...)`

Comment (cosenal) [inline waterbutler/providers/owncloud/provider.py]: actually here the best thing is probably a string concatenation with `+`

**Comment by felliott** [comment]
This PR has been superseded by #168.  Marking as closed.  Thanks for your hard work on this, @cosenal!  This was a huge help in getting the owncloud provider finished.  We'll start testing it soon, and we're hoping it'll go in shortly!

Cheers,
@felliott

**Comment by cosenal** [comment]
Great to hear that progress has been made on this!

**Pull request closed without merging**